src/shuffle.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Creates an array of shuffled values, using the
+ * [Fisher-Yates shuffle](https://en.wikipedia.org/wiki/Fisher-Yates_shuffle).
+ * @param items the array to shuffle
+ * @signature
+ *    R.shuffle(array)
+ * @example
+ *    R.shuffle(['one', 'two', 'three']) // => ['two', 'three', 'one']
+ * @category Array
+ */
+export function shuffle<T>(items: ReadonlyArray<T>): Array<T> {
+  const length = items.length
```

Comment:
I don't think we need to memoize the length prop of the array, I can't imagine this having an impact on perf over accessing items.length directly